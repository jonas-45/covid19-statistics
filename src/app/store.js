import { configureStore } from '@reduxjs/toolkit';
import leaguesSliceReducer from '../redux/leagues/leaguesSlice';

const store = configureStore({
  reducer: {
    leagues: leaguesSliceReducer,
  },
});

export default store;
