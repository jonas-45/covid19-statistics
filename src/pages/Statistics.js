import React from 'react';
import { Card } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight } from '@fortawesome/free-solid-svg-icons';

const Statistics = () => {
  const { covidStats } = useSelector((state) => state.covid);
  if (covidStats.length < 1) {
    return (
      <div>
        <p>Loading data, please wait...</p>
      </div>
    );
  }
  return (
    <div className="d-flex flex-wrap justify-content-center gap-3 p-3">
      {covidStats.map((info) => (
        <Card style={{ width: '18rem' }} key={info.id} className="bg-light p-2">
          <Card.Img variant="top" src={info.flag} style={{ width: '100%', height: '50%' }} />
          <Card.Body>
            <Card.Title>{info.country }</Card.Title>
            <div className="d-flex justify-content-between align-items-center">
              <Card.Subtitle>
                Deaths:
                {' '}
                {info.deaths}
              </Card.Subtitle>
              <Card.Subtitle><FontAwesomeIcon icon={faCircleArrowRight} size="2x" /></Card.Subtitle>
            </div>
            <Card.Text className='mt-3'>
              Polulation:
              {' '}
              {info.population}
            </Card.Text>
          </Card.Body>
        </Card>
      ))}
    </div>
  );
};

export default Statistics;
