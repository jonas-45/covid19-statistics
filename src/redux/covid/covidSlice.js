import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const url = 'https://disease.sh/v3/covid-19/countries';

export const getCovideStatistics = createAsyncThunk('covid/get-data', async () => {
  try {
    const response = await axios.get(url);
    const stats = await response.data.map((info) => ({
      id: info.countryInfo._id, //no-underscore-dangle
      updated: info.updated,
      flag: info.countryInfo.flag,
      country: info.country,
      cases: info.cases,
      tests: info.tests,
      deaths: info.deaths,
      population: info.population,
      recovered: info.recovered,
      active: info.active,
      critical: info.critical,
    }));
    return stats;
  } catch (error) {
    return false;
  }
});

const initialState = {
  covidStats: [],
  status: 'idle',
  error: null,
  search: '',
};

const covidSlice = createSlice({
  name: 'covidData',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(getCovideStatistics.fulfilled, (state, action) => ({
      ...state,
      covidStats: action.payload,
    }));
  },
});

export default covidSlice.reducer;
