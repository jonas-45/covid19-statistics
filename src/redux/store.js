import { configureStore } from '@reduxjs/toolkit';
import covidSliceReducer from './covid/covidSlice';

const store = configureStore({
  reducer: {
    covid: covidSliceReducer,
  },
});

export default store;
