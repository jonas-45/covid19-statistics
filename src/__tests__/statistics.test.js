import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import axios from 'axios';
import Statistics from '../pages/Statistics';
import store from '../redux/store';
import { getCovideStatistics } from '../redux/covid/covidSlice';

it('Country  page renders correctly', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Statistics />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Country  slicer and reducer check
describe('Countries redux state tests', () => {
  it('Should initially hold country data by default', () => {
    const state = store.getState().covid;
    expect(state.covidStats.length).toEqual(0);
  });

  it('Fetch covid data from API', async () => {
    const url = 'https://disease.sh/v3/covid-19/countries';
    const axiosSpy = jest.spyOn(axios, 'get');
    jest.setTimeout(90000);
    const dispatchSpy = jest.fn();
    await getCovideStatistics(url)(dispatchSpy);
    expect(axiosSpy).toHaveBeenCalledWith(url);
  });
});
