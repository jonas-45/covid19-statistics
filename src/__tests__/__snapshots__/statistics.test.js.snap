// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country  page renders correctly 1`] = `
<div
  className="text-center mt-5 pt-5"
>
  <p>
    Loading data, please wait...
  </p>
</div>
`;
